File: testing-job/src/main/java/com/example/testingjob/misc/WebClient.java
package com.example.testingjob.misc;

import org.sonarsource.scanner.api.internal.shaded.okhttp.OkHttpClient;
import org.sonarsource.scanner.api.internal.shaded.okhttp.Request;
import org.sonarsource.scanner.api.internal.shaded.okhttp.RequestBody;
import org.sonarsource.scanner.api.internal.shaded.okhttp.Response;

import java.io.IOException;

public class WebClient {


    /**
     *
     * @param method type of method to use
     * @param url url to consume the service
     * @param body body and parameters of the service
     * @return
     * @throws IOException
     */
    public Response makeRequest(String method, String url, RequestBody body) throws IOException {
        Response response;

        try {
            OkHttpClient client = new OkHttpClient().newBuilder()
                    .build();
            Request request = new Request.Builder()
                    .url(url)
                    .method(method, body)
                    .build();
            response = client.newCall(request).execute();

            return response;
        } catch (Exception e) {
            return null;
        }
    }

}

File: src/clients/WebClient.js
const makeRequest = async (url) => {
    let requestOptions = {
        method: 'GET',
        redirect: 'follow'
    };

    return await fetch(url, requestOptions)
        .then(response => response.text())
        .then(result => {
            return result;
        })
        .catch(error => console.log('::error', error));
}


const API = {
    async get(url) {
        return await makeRequest(url);
    }
};

export default API;
